test/unit.js

```diff
@@ -28,6 +28,47 @@ metatests.test('Expression chain', async (test) => {
   test.end();
 });
 
+metatests.test('Recursive expressions', async (test) => {
+  const sheet = new Sheet();
+  sheet.cells['A1'] = 100;
+  sheet.cells['B1'] = 2;
+  sheet.cells['C1'] = '= A1 *E1';
+  sheet.cells['D1'] = '=C1+8';
+  sheet.cells['E1'] = '=D1/2';
+
+  try {
+    test.strictSame(sheet.values['D1'], 208);
+  } catch (error) {
+    test.strictSame(error.constructor.name === 'Error', true);
+    test.strictSame(error.message, 'Recursive expression error');
+  }
+
+  try {
+    test.strictSame(sheet.values['E1'], 104);
+  } catch (error) {
+    test.strictSame(error.constructor.name === 'Error', true);
+    test.strictSame(error.message, 'Recursive expression error');
+  }
+  test.end();
+});
+
+metatests.test(
+  'Exit from recursive expression use Math.random',
+  async (test) => {
+    const sheet = new Sheet();
+    sheet.cells['A1'] = 100;
+    sheet.cells['B1'] = 2;
+    sheet.cells['C1'] = '=Math.round(Math.random()) ? A1 * E1 : 5';
```

Comment:
`Math.random` is not stable for use in unit tests. Of course, it is a low probability, that you will get 100 "one"s results in a row, but still, one of 2^100 test runs may fail

Reply:
Oh, ok, thanks
I used this test to show 'exit from recursive expression' (only math.random came to mind)
when somebody will say how assignment in expression is work to I fix previous test

Reply:
We can use `F1` as a counter, increment it each `C1` cell calculation triggered and exit recursion after certain limit

Reply:
> We can use `F1` as a counter, increment it each `C1` cell calculation triggered and exit recursion after certain limit

How we can change cell's value? @tshemsedinov 
[I written about it above](https://github.com/metarhia/metacalc/pull/25#discussion_r932482809)